lib/nfc/protocols/mifare_classic.c

```diff
@@ -882,12 +882,17 @@ bool mf_classic_emulator(
             break;
         }
 
-        if(cmd == 0x50 && plain_data[1] == 0x00) {
+        if(cmd == NFCA_CMD_HALT && plain_data[1] == 0x00) {
             FURI_LOG_T(TAG, "Halt received");
-            furi_hal_nfc_listen_sleep();
-            command_processed = true;
+            return false;
```

Comment:
Please, avoid return from the body of function. It's not safe if you allocate memory above return and exit from function before freeing that memory. Also it make code less readable.

I suggest rename variable `command_processed` -> `send_nack` and add new variable `need_reset`. Rework every place where you write `return` with filling these variables then call `break`. And return `need_reset` at the and of the function